Fix auth redirect

leanil opened the pull request:
Set default for env var: REACT_APP_LOGIN_URL

The client expected this env var, and went into a redirect loop
if it wasn't present. Adding it to transenv with the appropriate
default value and reading it from config fixes this problem.

leanil committed:
Make authentication error message visible

hleumas commented:
This seems to work, but it doesn't seem like the right solution. I'd rather understand why isn't the value of REACT_APP_LOGIN_URL present during the build step.

hleumas commented:
Ok, so I guess here is the solution, you just need to move the variable to `now.json:build.env`

```
// now.json file

{
  "build": {
    "env": {
      "REACT_APP_LOGIN": "/auth/login"
    }
  }
}
```

leanil commented:
> This seems to work, but it doesn't seem like the right solution. I'd rather understand why isn't the value of REACT_APP_LOGIN_URL present during the build step.

I was puzzled how `.env.production` was loaded into an env var. I didn't realize it's done by react during build, so thanks for pointing that out.

I've found out that now [ignores](https://github.com/zeit/now/commit/9755847855e33292bce6b8a0d5213023c9010790) the .env.production file since january, so that's why it broke now.
We could force it to upload the file, but I think your solution is the way to go.

leanil committed:
Set REACT_APP_LOGIN_URL as now build env var

Now started to ignore .env files at upload, so we need to set the
REACT_APP_LOGIN_URL variable in now.json to be present at build.

paldys commented on now.json:
```diff
@@ -15,5 +15,11 @@
+
+    "build": {
+        "env": {
+            "REACT_APP_LOGIN": "/auth/login"
```
Don't just copy-paste someone else's suggestion...

This should be `REACT_APP_LOGIN_URL`.
